plugin/path_creds.go

```diff
@@ -162,6 +162,32 @@ func (b *backend) generateAndReturnCreds(ctx context.Context, engineConf *config
 		return nil, err
 	}
 
+	var currentPassword, lastPassword string
+	if val, ok := previousCred["current_password"].(string); ok {
+		currentPassword = val
+	}
+
+	if val, ok := previousCred["last_password"].(string); ok {
+		lastPassword = val
+	}
+
+	wal := rotateCredentialEntry{
+		CurrentPassword:    currentPassword,
+		LastPassword:       lastPassword,
+		RoleName:           roleName,
+		TTL:                role.TTL,
+		ServiceAccountName: role.ServiceAccountName,
+		LastVaultRotation:  role.LastVaultRotation,
+	}
+
+	// If we can't persist the WAL, try to revert the password in AD.
+	walID, err := framework.PutWAL(ctx, storage, rotateCredentialWAL, wal)
+	if err != nil {
+		if err := b.client.UpdatePassword(engineConf.ADConf, role.ServiceAccountName, currentPassword); err != nil {
```

Comment:
I have the same concern as JM here, we can potentially setting this to empty if `currentPassword` is empty.

Reply:
I've adjusted the logic here a little. Basically is current password is unset, this means its the first time the creds are being read, so we should instead try to roll forward via WAL.

Reply:
Decided to not roll forward with WAL. If password is unset, we bail.